diff --git a/include/basalt/optical_flow/optical_flow.h b/include/basalt/optical_flow/optical_flow.h
index c4510cb..ad1b4fa 100644
--- a/include/basalt/optical_flow/optical_flow.h
+++ b/include/basalt/optical_flow/optical_flow.h
@@ -207,7 +207,7 @@ class OpticalFlowTyped : public OpticalFlowBase {
     E.resize(getNumCams());
     for (size_t i = 0; i < getNumCams(); i++) {
       Eigen::Matrix4d Ed;
-      SE3 T_i_j = calib.T_i_c[0].inverse() * calib.T_i_c[1];
+      SE3 T_i_j = calib.T_i_c[0].inverse() * calib.T_i_c[i];
       computeEssential(T_i_j.template cast<double>(), Ed);
       E[i] = Ed.cast<Scalar>();
     }
